### PR TITLE
Fixing: App terminated due to signal 5 instantiating RollbarPLCrashCollector

### LINKED ISSUE #66: App terminated due to signal 5 instantiating RollbarPLCrashCollector
**Describe the bug**
App will terminate with the following message:

`Message from debugger: Terminated due to signal 5`

This occurs on the line that instantiates a RollbarPLCrashCollector:

`let crashCollector = RollbarPLCrashCollector()`

**To Reproduce**
1. Add RollbarNotifier, RollbarPLCrashReporter, RollbarSwift cocopods v2.0.0-beta9 to Podfile, install
2. Add sample Swift code from docs into AppDelegate:
```
import RollbarNotifier
import RollbarPLCrashReporter
...
func application(_ application: UIApplication,
                     didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
    let config = RollbarConfig()
    config.destination.accessToken = "..."
    config.destination.environment = "..."
    let crashCollector = RollbarPLCrashCollector()
    Rollbar.initWithConfiguration(config, crashCollector: crashCollector)
    ....
}
```

**Expected behavior**
App will launch successfully.

**Screenshots**
N/A

**Rollbar-Apple SDK version: v2.0.0-beta9**

**Calling the SDK from (x-mark all suitable):**
[] Objective-C
[x] Swift
[] Other: 

**Runtime environment (please complete the following information as applicable):**
 - Device: MacBook Pro 16"/iOS Simulator iPhone 12
 - OS: iOS
 - OS version: 14.3

### PR DESCRIPTION
**Pull request by akornich**
## Description of the change

> Fixing #66 - App terminated due to signal 5 instantiating RollbarPLCrashCollector
> Updated demo apps to use the latest releases of the SDK

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix #66 

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: Demos/macosAppObjC/macosAppObjC/AppDelegate.m [hunk 1]
```diff
@@ -84,9 +84,9 @@ - (void)initRollbar {
 
     // optional crash reporter:
     id<RollbarCrashCollector> crashCollector =
-      nil;
+      //nil;
```
Comment (pawelsz-rb) [inline Demos/macosAppObjC/macosAppObjC/AppDelegate.m:87]: clean up ?

File: Demos/macosAppSwiftCocoapods/macosAppSwiftCocoapods/AppDelegate.swift [hunk 2]
```diff
@@ -22,29 +23,33 @@ class AppDelegate: NSObject, NSApplicationDelegate {
...
     func initRollbar() {
 
         // configure Rollbar:
-        let config = RollbarConfiguration.init();
-        
+        let config = RollbarConfig();
         //config.crashLevel = @"critical";
-        config.environment = "samples";
-        config.asRollbarConfig().customData = [ "someKey": "someValue", ];
+        config.destination.accessToken = "2ffc7997ed864dda94f63e7b7daae0f3";
```
Comment (pawelsz-rb) [inline Demos/macosAppSwiftCocoapods/macosAppSwiftCocoapods/AppDelegate.swift:43]: this looks like a real access token

File: RollbarPLCrashReporter/Sources/RollbarPLCrashReporter/RollbarPLCrashCollector.m [hunk 3]
```diff
@@ -20,11 +20,17 @@ +(void)initialize {
...
-            PLCrashReporterSignalHandlerTypeMach;
-#else
-            PLCrashReporterSignalHandlerTypeBSD;
-#endif
+        // NOTE: looking at comments around the PLCRASH_FEATURE_MACH_EXCEPTIONS
+        //       definition in PLCrashReporter and matching it with actual experiences
+        //       on the current Apple OSs releases we should only use
+        //       the PLCrashReporterSignalHandlerTypeBSD option for now.
+        //       Leaving the alternative commented out for the reference in the future.
+//#if PLCRASH_FEATURE_MACH_EXCEPTIONS
```
Comment (pawelsz-rb) [inline RollbarPLCrashReporter/Sources/RollbarPLCrashReporter/RollbarPLCrashCollector.m:28]: clean up?

File: Demos/macosAppObjC/macosAppObjC/AppDelegate.m (hunk 1 above)
Comment (akornich) [reply Demos/macosAppObjC/macosAppObjC/AppDelegate.m:87]: Nope :), I am reusing these alternatives during manual testing/demo.

File: Demos/macosAppSwiftCocoapods/macosAppSwiftCocoapods/AppDelegate.swift (hunk 2 above)
Comment (akornich) [reply Demos/macosAppSwiftCocoapods/macosAppSwiftCocoapods/AppDelegate.swift:43]: it is for the project dedicated to unit testing and demos of the Apple SDKs.

File: RollbarPLCrashReporter/Sources/RollbarPLCrashReporter/RollbarPLCrashCollector.m (hunk 3 above)
Comment (akornich) [reply RollbarPLCrashReporter/Sources/RollbarPLCrashReporter/RollbarPLCrashCollector.m:28]: Nope :), the comments above the commented out code explain why we keep it (at least for now).

File: Demos/macosAppSwiftCocoapods/macosAppSwiftCocoapods/AppDelegate.swift (hunk 2 above)
Comment (pawelsz-rb) [reply Demos/macosAppSwiftCocoapods/macosAppSwiftCocoapods/AppDelegate.swift:43]: well, usually it's not a good practice to make tokens as part of repo. you can read it from env var for demo purposes. 
for unit tests, I think you can put anything if you are testing the token setter

Comment (pawelsz-rb) [reply Demos/macosAppSwiftCocoapods/macosAppSwiftCocoapods/AppDelegate.swift:43]: besides , if this is a valid production token then somebody can use it to send many messages

Comment (akornich) [reply Demos/macosAppSwiftCocoapods/macosAppSwiftCocoapods/AppDelegate.swift:43]: In general, agree. But here many tests hit our API to have a full functionality stack test. I am also planning to add a bit later the way Rollbar.NET does - to have a test harness send test payloads and verify a full round of outcomes via our API reply stats that should match a unit test's expectations.  But I need to refactore this SDK more before I can get there.
So far this approach was working well for the users of iOS, Apple, and .NET SDKs. It also allows them to clone an SDK repo and run/debug fully working demoes/examples right away.

**Review by pawelsz-rb — APPROVED** [review]

**Pull request merged**